v0/destinations/iterable/transform.js

```diff
@@ -325,43 +333,202 @@ function process(event) {
   return processSingleMessage(event.message, event.destination);
 }
 
+function batchEvents(arrayChunks) {
+  const batchedResponseList = [];
+
+  // list of chunks [ [..], [..] ]
+  arrayChunks.forEach(chunk => {
+    const batchResponseList = [];
+    const metadata = [];
+
+    // extracting destination
+    // from the first event in a batch
+    const { destination } = chunk[0];
+    const { apiKey } = destination.Config;
+
+    let batchEventResponse = defaultBatchRequestConfig();
+
+    // Batch event into dest batch structure
+    chunk.forEach(ev => {
+      batchResponseList.push(ev.message.body.JSON);
+      metadata.push(ev.metadata);
+    });
+    // batching into bulkUpdate batch structure
+    if (chunk[0].message.endpoint.includes("/api/users")) {
+      batchEventResponse.batchedRequest.body.JSON = {
+        users: batchResponseList
+      };
+      batchEventResponse.batchedRequest.endpoint = IDENTIFY_BATCH_ENDPOINT;
+    } else {
+      // batching into bulkUpdate batch structure
+      batchEventResponse.batchedRequest.body.JSON = {
+        events: batchResponseList
+      };
+      batchEventResponse.batchedRequest.endpoint = TRACK_BATCH_ENDPOINT;
+    }
```

Comment:
Can we use a switch case here?

Reply:
I think adding a switch case here will look like this:
switch(chunk[0].message.endpoint.includes("/api/users"))
{
case "true: ......
case "false: ......
}
because there are different types of endpoints that contain "api/users" and "api/events"
Example: api/users/egisterDeviceToken, api/users/update, api/users/updateEmail........
I feel "if-else" statements are more understandable(as there are only 2 cases).

Reply:
Cool got your point